src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,131 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int START_CAPACITY = 10;
+    private static final int EMPTY = 0;
+    private static final double GROW = 1.5;
+    private int size;
+    private int capacity;
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[]) new Object[START_CAPACITY];
+        size = EMPTY;
+        capacity = START_CAPACITY;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == capacity) {
+            grow();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("IndexOutOfBounds");
```

Comment:
For better clarity, change the exception message from 'IndexOutOfBounds' to 'Index out of bounds for ArrayList'.